src/main/java/reposense/commits/CommitInfoAnalyzer.java

```diff
@@ -93,9 +101,64 @@ public static CommitResult analyzeCommit(CommitInfo commitInfo, RepoConfiguratio
             extractTagNames(tags);
         }
 
-        int insertion = getInsertion(statLine);
-        int deletion = getDeletion(statLine);
-        return new CommitResult(author, hash, date, messageTitle, messageBody, tags, insertion, deletion);
+        if (statLine.isEmpty()) { // empty commit, no files changed
+            return new CommitResult(author, hash, date, messageTitle, messageBody, tags, 0, 0, new HashMap<>());
+        }
+
+        String[] statInfos = statLine.split(NEW_LINE_SPLITTER);
+        String[] fileTypeContributions = Arrays.copyOfRange(statInfos, 0, statInfos.length - 1);
```

Comment:
Clarification, the array is copied here because you do not want to include the last element of `statInfos` ?

Reply:
yes. I did so that it is clearer